### PR TITLE
use preffix enable_ for extra packages

### PR DESCRIPTION
**Pull request by cristi1979**
- also $host = 'localhost' is always wrong, so use $::fqdn instead
- enable default rpm plugin

### CODE REVIEW

File: manifests/admin.pp [hunk 1]
```diff
@@ -93,14 +93,13 @@
...
-  $rpm                = $pulp::admin::params::rpm,
+  $enable_puppet      = $pulp::admin::params::puppet,
+  $enable_docker      = $pulp::admin::params::docker,
+  $enable_nodes       = $pulp::admin::params::nodes,
+  $enable_python      = $pulp::admin::params::python,
+  $enable_rpm         = $pulp::admin::params::rpm,
   $puppet_upload_working_dir = $pulp::admin::params::puppet_upload_working_dir,
-  $puppet_upload_chunk_size  = $pulp::admin::params::puppet_upload_chunk_size,
-) inherits pulp::admin::params {
+  $puppet_upload_chunk_size  = $pulp::admin::params::puppet_upload_chunk_size,) inherits pulp::admin::params {
```
Comment (ehelms) [inline manifests/admin.pp]: Why move this closing paren and inherits to the same line as the last param?

**Comment by cristi1979** [comment]
fixed geppetto autoformat

**Comment by ehelms** [comment]
This is looking good - mind squashing commits?

**Comment by cristi1979** [comment]
- squashed
- I also made enable_rpm = true by default

File: manifests/admin.pp [hunk 2]
```diff
@@ -93,14 +93,14 @@
...
-  $puppet             = $pulp::admin::params::puppet,
-  $docker             = $pulp::admin::params::docker,
-  $nodes              = $pulp::admin::params::nodes,
-  $python             = $pulp::admin::params::python,
-  $rpm                = $pulp::admin::params::rpm,
+  $enable_puppet      = $pulp::admin::params::puppet,
+  $enable_docker      = $pulp::admin::params::docker,
+  $enable_nodes       = $pulp::admin::params::nodes,
+  $enable_python      = $pulp::admin::params::python,
+  $enable_rpm         = $pulp::admin::params::rpm,
```
Comment (ehelms) [inline manifests/admin.pp]: Any reason for the parameters and default params to mimic each other in name?

File: manifests/admin.pp [hunk 3]
```diff
@@ -93,14 +93,14 @@
...
-  $puppet             = $pulp::admin::params::puppet,
-  $docker             = $pulp::admin::params::docker,
-  $nodes              = $pulp::admin::params::nodes,
-  $python             = $pulp::admin::params::python,
-  $rpm                = $pulp::admin::params::rpm,
+  $enable_puppet      = $pulp::admin::params::enable_puppet,
+  $enable_docker      = $pulp::admin::params::enable_docker,
+  $enable_nodes       = $pulp::admin::params::enable_nodes,
+  $enable_python      = $pulp::admin::params::enable_python,
+  $enable_rpm         = $pulp::admin::params::enable_rpm,
```
Comment (ehelms) [inline manifests/admin.pp]: Not a requirement for the PR, but you cold consider adding `validate_bool` for each to enforce the type

**Comment by cristi1979** [comment]
I've also added 2 new packages to the puppet installation: 'pulp-puppet-tools', 'python-pulp-puppet-common'. The second one is required because of bug 1105673

**Comment by cristi1979** [comment]
all tests for ruby 1.8.7 failed with an exception. Since I couldn't find how to restart the travis jobs, I will close/reopen this PR.

**Comment by ekohl** [comment]
Test failures are due to https://github.com/rspec/rspec-core/issues/1864

**Comment by stbenjam** [comment]
https://github.com/Katello/puppet-pulp/pull/47 fixes that

**Comment by ekohl** [comment]
Nitpick: it's a prefix, not a suffix.

**Comment by ehelms** [comment]
This looks good to me. @ekohl what say ye

File: manifests/admin/params.pp [hunk 4]
```diff
@@ -1,7 +1,7 @@
 # Pulp Admin Params
 class pulp::admin::params {
   $version            = 'installed'
-  $host               = 'localhost'
+  $host               = $::fqdn
```
Comment (ekohl) [inline manifests/admin/params.pp]: This change appears to be unrelated and I don't know pulp well enough to know if it makes a difference.

**Comment by ekohl** [comment]
I think it looks good. This does mean a major version bump of the module, but that can be ok. I don't know if a version shipped in the installer with these parameters, but that may also be worth considering and testing how it'll work for upgrading users.

**Comment by cristi1979** [comment]
Template was changed to use scope[] instead of scope.lookupvar

**Comment by ehelms** [comment]
@ekohl We cut a stable branch of 0.1 before these changes so that we could bring them all in and eventually release a 1.0.0. Does that sounds reasonable? Anything else you'd suggest?

**Comment by ekohl** [comment]
@ehelms given you're in the 0.x series, I think that's fine. People should know a 0.x version can break the API with every release.

**Comment by ehelms** [comment]
ACK - thanks @cristi1979
